Logback initialization failed on Windows Gradle build

### Expected Behavior

Gradle build completes successfully without any errors on Windows as well as on Linux.

### Actual Behaviour

I had a logback initialization error caused by micronaut-test-resources.

<details>
<summary>:ng: gradlew build logs (Windows 10 21H1 + OpenJDK Temurin 11.0.16)</summary>

```
> Configure project :
[native-image-plugin] Instrumenting task with the native-image-agent: test
[native-image-plugin] Instrumenting task with the native-image-agent: testNativeImage
22:18:30,805 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
22:18:30,805 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.xml]
22:18:30,821 |-INFO in ch.qos.logback.core.ConsoleAppender[null] - Enabling JANSI WindowsAnsiOutputStream for the console.
22:18:30,821 |-WARN in ch.qos.logback.core.ConsoleAppender[null] - Failed to create WindowsAnsiOutputStream. Falling back on the default stream. ch.qos.logback.core.util.DynamicClassLoadingException: Failed to instantiate type org.fusesource.jansi.WindowsAnsiOutputStream
        at ch.qos.logback.core.util.DynamicClassLoadingException: Failed to instantiate type org.fusesource.jansi.WindowsAnsiOutputStream
        at      at ch.qos.logback.core.util.OptionHelper.instantiateByClassNameAndParameter(OptionHelper.java:68)
        at      at ch.qos.logback.core.util.OptionHelper.instantiateByClassNameAndParameter(OptionHelper.java:39)
        at      at ch.qos.logback.core.ConsoleAppender.getTargetStreamForWindows(ConsoleAppender.java:88)
        at      at ch.qos.logback.core.ConsoleAppender.start(ConsoleAppender.java:79)
        at      at io.micronaut.testresources.server.StaticLogbackConfiguration.configure(StaticLogbackConfiguration.java:26)
        at      at ch.qos.logback.classic.util.ContextInitializer.autoConfig(ContextInitializer.java:140)
        at      at org.slf4j.impl.StaticLoggerBinder.init(StaticLoggerBinder.java:84)
        at      at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:55)
        at      at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
        at      at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
        at      at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:417)
        at      at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:362)
        at      at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:388)
        at      at io.micronaut.testresources.server.Application.<clinit>(Application.java:37)
Caused by: java.lang.ClassNotFoundException: org.fusesource.jansi.WindowsAnsiOutputStream
        at      at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at      at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        at      at ch.qos.logback.core.util.OptionHelper.instantiateByClassNameAndParameter(OptionHelper.java:55)
        at      ... 13 common frames omitted

22:18:31.039 [ForkJoinPool.commonPool-worker-5] INFO  i.m.t.e.TestResourcesResolverLoader - Loaded 2 test resources resolvers: io.micronaut.testresources.mysql.MySQLTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvider
22:18:31.368 [main] INFO  i.m.testresources.server.Application - A Micronaut Test Resources server is listening on port 50985, started in 531msourcesService

> Task :compileTestJava
Note: Creating bean classes for 1 type elements
22:18:36.826 [default-nioEventLoopGroup-1-2] INFO  i.m.t.e.TestResourcesResolverLoader - Loaded 2 test resources resolvers: io.micronaut.testresources.mysql.MySQLTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvideruting test com.example.DemoTest
22:18:37.137 [default-nioEventLoopGroup-1-7] INFO  o.t.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')Executing test com.example.DemoTest
22:18:37.137 [default-nioEventLoopGroup-1-7] INFO  i.m.t.testcontainers.TestContainers - Starting test container mysql
22:18:37.153 [default-nioEventLoopGroup-1-7] INFO  o.t.d.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.NpipeSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
22:18:37.352 [default-nioEventLoopGroup-1-7] INFO  o.t.d.DockerClientProviderStrategy - Found Docker environment with local Npipe socket (npipe:////./pipe/docker_engine)
22:18:37.352 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - Docker host IP address is localhost
22:18:37.363 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - Connected to docker:
  Server Version: 20.10.17
  API Version: 1.41
  Operating System: Docker Desktop
  Total Memory: 25588 MB
22:18:37.377 [default-nioEventLoopGroup-1-7] INFO  docker[testcontainers/ryuk:0.3.3] - Creating container for image: testcontainers/ryuk:0.3.3
22:18:37.520 [default-nioEventLoopGroup-1-7] INFO  o.t.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop) does not have credentials for https://index.docker.io/v1/
22:18:37.579 [default-nioEventLoopGroup-1-7] INFO  docker[testcontainers/ryuk:0.3.3] - Container testcontainers/ryuk:0.3.3 is starting: 9d0ea26d89ec0cbc8385783d47564db92e1f8b1f5395f87ab5cd84708a91a3a9
22:18:37.933 [default-nioEventLoopGroup-1-7] INFO  docker[testcontainers/ryuk:0.3.3] - Container testcontainers/ryuk:0.3.3 started in PT0.5649302Sd
22:18:37.938 [default-nioEventLoopGroup-1-7] INFO  o.t.utility.RyukResourceReaper - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
22:18:37.938 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - Checking the system...
22:18:37.938 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - ?? Docker server version should be at least 1.6.0
22:18:37.947 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Creating container for image: mysql:latest
22:18:38.006 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Container mysql:latest is starting: 7ddf9dfac371fd39b32431d43a7f5244dda9055d6c5187efde40ce9e71c1f4b3
22:18:39.199 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Waiting for database connection to become available at jdbc:mysql://localhost:51096/test using query 'SELECT 1'
22:18:45.590 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Container is started (JDBC URL: jdbc:mysql://localhost:51096/test)mpleted
22:18:45.591 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Container mysql:latest started in PT7.6518007S

BUILD SUCCESSFUL in 22s
15 actionable tasks: 15 executed
```
</details>

<details>
<summary>:ng: gradlew build logs (Windows 10 21H1 + Oracle JDK 17.0.4)</summary>

```
> Configure project :
[native-image-plugin] Instrumenting task with the native-image-agent: test
[native-image-plugin] Instrumenting task with the native-image-agent: testNativeImage
21:58:40,135 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
21:58:40,135 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.xml]
21:58:40,167 |-INFO in ch.qos.logback.core.ConsoleAppender[null] - Enabling JANSI WindowsAnsiOutputStream for the console.
21:58:40,167 |-WARN in ch.qos.logback.core.ConsoleAppender[null] - Failed to create WindowsAnsiOutputStream. Falling back on the default stream. ch.qos.logback.core.util.DynamicClassLoadingException: Failed to instantiate type org.fusesource.jansi.WindowsAnsiOutputStream
        at ch.qos.logback.core.util.DynamicClassLoadingException: Failed to instantiate type org.fusesource.jansi.WindowsAnsiOutputStream
        at      at ch.qos.logback.core.util.OptionHelper.instantiateByClassNameAndParameter(OptionHelper.java:68)
        at      at ch.qos.logback.core.util.OptionHelper.instantiateByClassNameAndParameter(OptionHelper.java:39)
        at      at ch.qos.logback.core.ConsoleAppender.getTargetStreamForWindows(ConsoleAppender.java:88)
        at      at ch.qos.logback.core.ConsoleAppender.start(ConsoleAppender.java:79)
        at      at io.micronaut.testresources.server.StaticLogbackConfiguration.configure(StaticLogbackConfiguration.java:26)
        at      at ch.qos.logback.classic.util.ContextInitializer.autoConfig(ContextInitializer.java:140)
        at      at org.slf4j.impl.StaticLoggerBinder.init(StaticLoggerBinder.java:84)
        at      at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:55)
        at      at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
        at      at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
        at      at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:417)
        at      at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:362)
        at      at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:388)
        at      at io.micronaut.testresources.server.Application.<clinit>(Application.java:37)
Caused by: java.lang.ClassNotFoundException: org.fusesource.jansi.WindowsAnsiOutputStream
        at      at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at      at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
        at      at ch.qos.logback.core.util.OptionHelper.instantiateByClassNameAndParameter(OptionHelper.java:55)
        at      ... 13 common frames omitted

21:58:40.496 [ForkJoinPool.commonPool-worker-5] INFO  i.m.t.e.TestResourcesResolverLoader - Loaded 2 test resources resolvers: io.micronaut.testresources.mysql.MySQLTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvider
21:58:40.758 [main] INFO  i.m.testresources.server.Application - A Micronaut Test Resources server is listening on port 50711, started in 583msourcesService

> Task :compileTestJava
Note: Creating bean classes for 1 type elements
21:58:45.753 [default-nioEventLoopGroup-1-2] INFO  i.m.t.e.TestResourcesResolverLoader - Loaded 2 test resources resolvers: io.micronaut.testresources.mysql.MySQLTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvideruting test com.example.DemoTest
21:58:46.044 [default-nioEventLoopGroup-1-7] INFO  o.t.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')Executing test com.example.DemoTest
21:58:46.059 [default-nioEventLoopGroup-1-7] INFO  i.m.t.testcontainers.TestContainers - Starting test container mysql
21:58:46.059 [default-nioEventLoopGroup-1-7] INFO  o.t.d.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.NpipeSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
21:58:46.254 [default-nioEventLoopGroup-1-7] INFO  o.t.d.DockerClientProviderStrategy - Found Docker environment with local Npipe socket (npipe:////./pipe/docker_engine)
21:58:46.254 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - Docker host IP address is localhost
21:58:46.266 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - Connected to docker:
  Server Version: 20.10.17
  API Version: 1.41
  Operating System: Docker Desktop
  Total Memory: 25588 MB
21:58:46.280 [default-nioEventLoopGroup-1-7] INFO  docker[testcontainers/ryuk:0.3.3] - Creating container for image: testcontainers/ryuk:0.3.3leted
21:58:46.427 [default-nioEventLoopGroup-1-7] INFO  o.t.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop) does not have credentials for https://index.docker.io/v1/
21:58:46.479 [default-nioEventLoopGroup-1-7] INFO  docker[testcontainers/ryuk:0.3.3] - Container testcontainers/ryuk:0.3.3 is starting: 292b25051690aa70118d9fd59c19a2e1999b471cc5063b1d7a9da53286eba104
21:58:46.896 [default-nioEventLoopGroup-1-7] INFO  docker[testcontainers/ryuk:0.3.3] - Container testcontainers/ryuk:0.3.3 started in PT0.6258872Sd
21:58:46.900 [default-nioEventLoopGroup-1-7] INFO  o.t.utility.RyukResourceReaper - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
21:58:46.901 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - Checking the system...
21:58:46.901 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - ?? Docker server version should be at least 1.6.0
21:58:46.914 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Creating container for image: mysql:latest
21:58:46.971 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Container mysql:latest is starting: 336cb078f3829d70f0411118b3eb7516738a2813879eff36ed3264d95e4ccf65
21:58:48.708 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Waiting for database connection to become available at jdbc:mysql://localhost:50722/test using query 'SELECT 1'
21:58:54.384 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Container is started (JDBC URL: jdbc:mysql://localhost:50722/test)mpleted
21:58:54.384 [default-nioEventLoopGroup-1-7] INFO  docker[mysql:latest] - Container mysql:latest started in PT7.4817511S

BUILD SUCCESSFUL in 21s
15 actionable tasks: 15 executed
```
</details>

<details>
<summary>:ok: gradlew build logs (Ubuntu 20.04.4 LTS + OpenJDK 17.0.4)</summary>

```
> Configure project :
[native-image-plugin] Instrumenting task with the native-image-agent: test
[native-image-plugin] Instrumenting task with the native-image-agent: testNativeImage
22:01:44.029 [ForkJoinPool.commonPool-worker-6] INFO  i.m.t.e.TestResourcesResolverLoader - Loaded 2 test resources resolvers: io.micronaut.testresources.mysql.MySQLTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvider
22:01:44.341 [main] INFO  i.m.testresources.server.Application - A Micronaut Test Resources server is listening on port 46481, started in 562msvice

> Task :compileTestJava
Note: Creating bean classes for 1 type elements
22:02:10.836 [default-nioEventLoopGroup-1-2] INFO  i.m.t.e.TestResourcesResolverLoader - Loaded 2 test resources resolvers: io.micronaut.testresources.mysql.MySQLTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvider
22:02:11.648 [default-nioEventLoopGroup-1-7] INFO  o.t.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
22:02:11.651 [default-nioEventLoopGroup-1-7] INFO  i.m.t.testcontainers.TestContainers - Starting test container mysql
22:02:11.665 [default-nioEventLoopGroup-1-7] INFO  o.t.d.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
22:02:11.972 [default-nioEventLoopGroup-1-7] INFO  o.t.d.DockerClientProviderStrategy - Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
22:02:11.973 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - Docker host IP address is localhost
22:02:11.982 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - Connected to docker:
  Server Version: 20.10.17
  API Version: 1.41ompleted
  Operating System: Docker Desktop
  Total Memory: 25588 MB
22:02:12.000 [default-nioEventLoopGroup-1-7] INFO  🐳 [testcontainers/ryuk:0.3.3] - Creating container for image: testcontainers/ryuk:0.3.3
22:02:22.351 [default-nioEventLoopGroup-1-7] INFO  o.t.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop.exe) does not have credentials for https://index.docker.io/v1/
22:02:22.416 [default-nioEventLoopGroup-1-7] INFO  🐳 [testcontainers/ryuk:0.3.3] - Container testcontainers/ryuk:0.3.3 is starting: 74b9b7e8a5b080ae86620c871611ce2246343221dda92000ef09c19ae46e8b7d
22:02:22.777 [default-nioEventLoopGroup-1-7] INFO  🐳 [testcontainers/ryuk:0.3.3] - Container testcontainers/ryuk:0.3.3 started in PT10.787395579Se.DemoTest
22:02:22.782 [default-nioEventLoopGroup-1-7] INFO  o.t.utility.RyukResourceReaper - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
22:02:22.782 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - Checking the system... :test > Executing test com.example.DemoTest
22:02:22.783 [default-nioEventLoopGroup-1-7] INFO  o.testcontainers.DockerClientFactory - ✔︎ Docker server version should be at least 1.6.0
22:02:22.792 [default-nioEventLoopGroup-1-7] INFO  🐳 [mysql:latest] - Creating container for image: mysql:latest
22:02:22.853 [default-nioEventLoopGroup-1-7] INFO  🐳 [mysql:latest] - Container mysql:latest is starting: fcef7daad95f3ceffbe9c7596973e94f5b7db4a79e624279914f572f4846d96a
22:02:23.192 [default-nioEventLoopGroup-1-7] INFO  🐳 [mysql:latest] - Waiting for database connection to become available at jdbc:mysql://localhost:50786/test using query 'SELECT 1'
22:02:29.672 [default-nioEventLoopGroup-1-7] INFO  🐳 [mysql:latest] - Container is started (JDBC URL: jdbc:mysql://localhost:50786/test)
22:02:29.672 [default-nioEventLoopGroup-1-7] INFO  🐳 [mysql:latest] - Container mysql:latest started in PT6.888017329S

BUILD SUCCESSFUL in 53s
15 actionable tasks: 15 executed
```
</details>

### Steps To Reproduce

1. Create a demo project with the mysql feature using Micronaut Launch.
2. Run the gradle build task.

### Environment Information

:ng: 1. Windows 10 21H1 + OpenJDK Temurin 11.0.16
:ng: 2. Windows 10 21H1 + Oracle JDK 17.0.4
:ok: 3. Ubuntu 20.04.4 LTS + OpenJDK 17.0.4

### Example Application

_No response_

### Version

Micronaut 3.6.0, Micronaut Test Resources Plugin 3.5.1

--- a/test-resources-server/src/main/resources/logback.xml
+++ b/test-resources-server/src/main/resources/logback.xml
@@ -1,7 +1,7 @@
 <configuration>
 
     <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
-        <withJansi>true</withJansi>
+        <withJansi>false</withJansi>
         <!-- encoders are assigned the type
              ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
         <encoder>
@@ -10,9 +10,9 @@
     </appender>
 
     <root level="info">
-        <appender-ref ref="STDOUT" />
+        <appender-ref ref="STDOUT"/>
     </root>
-   <logger name="io.micronaut.runtime.Micronaut" level="warn" additivity="false">
+    <logger name="io.micronaut.runtime.Micronaut" level="warn" additivity="false">
         <appender-ref ref="STDOUT"/>
     </logger>
 </configuration>

